packages/protocol/src/abstracts/BaseRouter.sol

```diff
@@ -278,13 +309,12 @@ abstract contract BaseRouter is SystemAccessControl, IRouter {
           bytes memory requestorCalldata
```

Comment:
Missing validation for the flash amount and asset, which would allow users to sweep tokens with fake flash loans.
Consider changing the above logic to, instead of receiving `requestorCalldata`, receive `innerActions` and `innerArgs`, and then build the requestor calldata (so there is also no need to slice).
Something like:
```solidity
bytes memory requestorCalldata_ = abi.encodeWithSelector(this.xBundleFlashloan.selector, innerActions, innerArgs, asset, flashAmount);
```

Reply:
>Missing validation for the flash amount and asset, which would allow users to sweep tokens with fake flash loans.
At what position the validation is suggested? At reentry call `xBundleFlashloan(...0?`, or just after decoding args in the `IRouter.Action.Flashloan` inside the `_internalBundle(...)` ?

Reply:
RequestorCalldata is now encoded in the `_bundleInternal(...)` as suggested and therefore removed the slicing operations.

Reply:
looks good now